cpp/src/binaryop/compiled/equality_ops.cu

```diff
@@ -37,7 +37,6 @@ void dispatch_equality_op(mutable_column_device_view& outd,
       switch (op) {
       case binary_operator::EQUAL:         device_type_dispatcher<ops::Equal>{outd, lhsd, rhsd, is_lhs_scalar, is_rhs_scalar, common_dtype}(i); break;
       case binary_operator::NOT_EQUAL:     device_type_dispatcher<ops::NotEqual>{outd, lhsd, rhsd, is_lhs_scalar, is_rhs_scalar, common_dtype}(i); break;
```

Comment:
Does it still make sense to dispatch `EQUAL` and `NOT_EQUAL` in this way? The code claims just above this line that
```
// Similar enabled template types should go together (better performance)
```
but that seems not to be true for the case of `NULL_EQUALS` (which is why this PR was created).

Why would this affect `NULL_EQUALS` but not the other dispatches? Additional handling of bitmask data that doesn't occur for the other two?

Reply:
It still might affect these 2 dispatch as well. will benchmark these too.
When I worked on compiled binary ops, I tried several combinations of operators to fuse. 
Combining Less, LessEqual,  Greater, GreaterEqual faced exact same performance issue shown here. Hence, this quick fix made sense.

Fusing was done for couple of reasons - 1. compile time, 2. performance. Of all the combinations I tried, this only had better  performance and lesser compile time, though performance benefit was much less.

Reply:
> cuobjdump resource usage
> 
> equality_ops.cu (EQUALS, NOT_EQUAL, NULL_EQUALS): REG:42 STACK:208 SHARED:0 LOCAL:0 CONSTANT[0]:592 CONSTANT[2]:4832 TEXTURE:0 SURFACE:0 SAMPLER:0 equality_ops.cu ((EQUALS, NOT_EQUAL): REG:40 STACK:0 SHARED:0 LOCAL:0 CONSTANT[0]:592 CONSTANT[2]:3208 TEXTURE:0 SURFACE:0 SAMPLER:0 NullEquals.cu (NULL_EQUALS): REG:42 STACK:0 SHARED:0 LOCAL:0 CONSTANT[0]:576 CONSTANT[2]:1600 TEXTURE:0 SURFACE:0 SAMPLER:0
> 
> I could see difference between constant memory and STACK. Looking into ptx and profiling too.

My suspicion from those values is that splitting the `EQUAL` / `NOT_EQUAL` dispatcher won't improve performance any further, and that it is probably fine to leave as-is (go ahead and post the results if you choose to test performance with this split apart).

From what I saw with the AST code, I would guess that the stack frame with `NULL_EQUALS` is killing the performance due to local memory transfer.